### PR TITLE
Prevent autostart from re-enabling practice mode after the user has explicity disabled it

### PR DESCRIPTION
**Pull request by Lidbetter**

### CODE REVIEW

**Comment by Lidbetter** [comment]
I may also need to update compaison checks - not sure how sourcepawn handles that (having issues with building locally to check for errors), will address if needed when I am able to build and test locally.

Is this general approach acceptable?

File: scripting/practicemode.sp [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #pragma semicolon 1
 #pragma newdecls required
 
-bool g_InPracticeMode = false;
+int g_InPracticeMode = -1;
```
Comment (splewis) [inline scripting/practicemode.sp:25]: I wouldn't suggest doing it this way, this leads to some fairly confusing code downstream: it's not obvious to a reader what `g_InPracticeMode=-1` means without diving into all the uses of `g_InPracticeMode`.

I would suggest instead adding a new variable, `bool g_AutostartDisabledThisMap` or such (defaulting to false), that gets set to true once the user has exited practicemode, and then checking it inside `CheckAutoStart`.

Alternatively, a 3-state enum could be defined and used the way you're doing it now, but that feels unnecessary to me.

Comment (Lidbetter) [reply scripting/practicemode.sp]: done, went with `g_PracticeModeCanBeAutoStarted`

**Comment by Lidbetter** [comment]
I have not tested as I only have my macbook with me (no linux vm and a very slow connection)
but I think it should be close, I will compile and test when I am back at my desktop sometime in the next few days

**Comment by mrc4tt** [comment]
> I have not tested as I only have my macbook with me (no linux vm and a very slow connection) but I think it should be close, I will compile and test when I am back at my desktop sometime in the next few days

Hi @Lidbetter.

I did test it, and it works (if I disconnect from the gameserver) - What with, if you switch map - where practicemode is still loaded (and not unloaded), so clients will not get "grenades" on change-map, but only at "reconnect". What with change-map?

**Comment by Lidbetter** [comment]
This is ready for review

**Review by ksgoescoding — APPROVED** [review]

**Review by ksgoescoding — APPROVED** [review]

**Comment by Lidbetter** [comment]
@splewis bumping incase you missed earler review request

**Pull request merged**